### ISSUE TITLE
archive-async in combination with archive-max-mb doesn't work as documented

### ISSUE THREAD

**Issue by terrorobe** [pgmasters/backrest#84]
Currently, ``archive-async`` will propagate a non-zero exit code to postgres when an archive-stop file has been created, causing WAL files to pile up in ``pg_xlog``:

```
postgres@prd-pgbackslave-02-tor ~ $ /opt/backrest/bin/pg_backrest.pl --stanza=backslave-02 archive-push pg_xlog/000000010000000000000071
2015-05-27 09:26:34.515 T00  ERROR: archive stop file (/data/postgres_backups/lock/backslave-02-archive.stop) exists , discarding 000000010000000000000071
2015-05-27 09:26:34.516 T00  ERROR: process terminated on signal or exception, 0 threads stopped
Undefined subroutine &BackRest::Archive::remote_exit called at /opt/backrest/bin/../lib/BackRest/Archive.pm line 360.
 at /opt/backrest/bin/pg_backrest.pl line 285.
postgres@prd-pgbackslave-02-tor ~ $ echo $?
255
postgres@prd-pgbackslave-02-tor ~ $
```

**Comment by dwsteele** [comment]
I'll look at this today.

**Comment by dwsteele** [comment]
Fixed in 3083e5f.
